controllers/policystatus_utils.go

```diff
@@ -80,23 +80,28 @@ func SetPolicyConfigurationCondition(policyServerConfigMap *corev1.ConfigMap, po
 	}
 }
 
+func isLatestReplicaSetFromPolicyServerDeployment(replicaSet *appsv1.ReplicaSet, policyServerDeployment *appsv1.Deployment) bool {
+	return replicaSet.Annotations[constants.KubernetesRevisionAnnotation] == policyServerDeployment.Annotations[constants.KubernetesRevisionAnnotation] &&
+		replicaSet.Annotations[constants.PolicyServerDeploymentConfigVersionAnnotation] == policyServerDeployment.Annotations[constants.PolicyServerDeploymentConfigVersionAnnotation]
+}
+
 func isPolicyUniquelyReachable(ctx context.Context, apiReader client.Reader, policyServerDeployment *appsv1.Deployment) bool {
 	replicaSets := appsv1.ReplicaSetList{}
-	if err := apiReader.List(ctx, &replicaSets, client.InNamespace(policyServerDeployment.Namespace)); err != nil {
+	if err := apiReader.List(ctx, &replicaSets, client.MatchingLabels{constants.PolicyServerLabelKey: policyServerDeployment.Labels[constants.PolicyServerLabelKey]}); err != nil {
 		return false
 	}
 	podTemplateHash := ""
-	for _, replicaSet := range replicaSets.Items {
-		if replicaSet.Annotations[constants.PolicyServerDeploymentConfigVersionAnnotation] == policyServerDeployment.Annotations[constants.PolicyServerDeploymentConfigVersionAnnotation] {
-			podTemplateHash = replicaSet.Labels[appsv1.DefaultDeploymentUniqueLabelKey]
+	for index := range replicaSets.Items {
```

Comment:
Why not loop with two vars as before, ignoring the index?

Reply:
Because our linter complains about that: ` Error: G601: Implicit memory aliasing in for loop. (gosec)`

https://github.com/kubewarden/kubewarden-controller/runs/5838848906?check_suite_focus=true

Reply:
Phew... good. Thanks :)